src/io/flutter/logging/FlutterLogEntryParser.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.logging;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.openapi.util.Key;
+import org.dartlang.vm.service.element.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+import static io.flutter.logging.FlutterLog.LOGGING_STREAM_ID;
+
+public class FlutterLogEntryParser {
+
+  // Known entry categories.
+  public static final String LOG_CATEGORY = "LOG";
```

Comment:
Perhaps use lowercaseCamelCase for log categories, with subcategories separated by periods? So, something like `flutter.frame`, `flutter.someOtherEvent`, `stdio.stdout`, ...

That pattern would also give framework and app developers easy places to plug in (`flutter.redux`, `fooApp.pageOpen`, ...)